### ISSUE TITLE
Update localadmin for South Africa

### ISSUE THREAD

**Issue by sindile** [whosonfirst-data/whosonfirst-data#1991]
It seems that the localadmin data for South Africa is outdated.
For example:
Nxuba - 1125271139
Nkonkobe - 1125278467
The above were merged. How do I send the new boundary data?

**Comment by missinglink** [comment]
Hi @sindile, could you please provide some links to an authoritative + openly licensed data source which could be used as a replacement?

**Comment by sindile** [comment]
Hi @missinglink
The institution responsible for boundary redetermination is the https://www.demarcation.org.za/#
The smallest unit is the ward, which are agreggated into municipalities:
wards - https://dataportal-mdb-sa.opendata.arcgis.com/datasets/9092c804a70a49efa40932082eacdb55_0
local municipality - https://dataportal-mdb-sa.opendata.arcgis.com/datasets/f676be280ea94648999d176d58a5e772_0
district municipality - https://dataportal-mdb-sa.opendata.arcgis.com/datasets/aaba805ffd254cb7bdf985db8eb7e37f

The above generally change every five years.

**Comment by missinglink** [comment]
Thanks @sindile, it's not clear how this data is licensed and if that license allows it to be included in whosonfirst.

<img width="237" alt="Screenshot 2022-01-18 at 17 08 54" src="https://user-images.githubusercontent.com/738069/149974562-8715497c-eeeb-475d-8788-92a812275a7d.png">

Are you able to clarify how this data is licensed?

**Comment by sindile** [comment]
Hi @missinglink 
Generally in South Africa data collected by public institutions is public data.
I can find out from the Municipal Demarcation Board and report back.
I might as well add that the data currently used in WOF would have come from the Municipal Demarcation Board.

**Comment by nvkelso** [comment]
@stepps00 can you take a look, please?

**Comment by sindile** [comment]
Hi @missinglink @nvkelso @stepps00 
Here is the clarification from the Municipal Demarcation Board
![image](https://user-images.githubusercontent.com/1027916/150472821-6e18f00a-b672-4d5a-9feb-44b6669c6fd7.png)

**Comment by stepps00** [comment]
This data looks great.. thanks for following up with license information, @sindile 

I'm unable to find a license through the dataportal links above, but I will take a closer look once we start to import. The links provide enough to update WOF records at all placetypes in South Africa. I'll set aside time next week to compare features and reply here with an update.

**Comment by sindile** [comment]
I am glad to have been of assistance and looking forward to the updates.

**Comment by stepps00** [comment]
I spent a bit of time reviewing the source data from the Municipal Demarcation Board, comparing it with existing WOF records. I'd propose the following changes per placetype:

- **Country**
	- No changes
- **Region (Provinces)**
	- No changes
	- Dissolving the district municipalities on the "DISTRICT" attribute results in the same regions currently in WOF
- **County (District Municipalities)**
	- No changes
- **Localadmin (Local Municipalities)**
	- Replace existing WOF Point geometries with geometries from Local Municipalities file
	- In some cases, WOF records will be superseded (see examples in initial issue comment)
- **Locality**
	- Clip geometries to parent localadmin geometries, when necessary
	- Review large cities, potentially digitize new geometries for a handful of places
- **Neighbourhood**
	- Review neighbourhoods in large cities, potentially digitize new geometries for a handful of places

Overall, the biggest changes here would be to the localadmin placetype. We could curate better data at the locality and neighbourhood placetypes, too. 

@sindile - you mentioned the regions will need to be updated, but I'm actually seeing the Province data in these files matches the region data currently in WOF. Do you have examples where a WOF region needs an update?

Also, I see unique codes for features in the Local Municipalities file, but I'm unable to find a Name attribute. Any idea where we'd find official names for these places?

**Comment by nvkelso** [comment]
Sounds like a good plan.

> Clip geometries to parent localadmin geometries, when necessary

Are the localities often crossing the localadmin boundaries? Is this a multi-hierarchy issue? 

If the localities are primary in terms of government administration then I think multi-hierarchy with multiple localadmin.

Generally WOF tries to treat localities (not localadmin) as the "essential question", so I'd hesitate to cut up the localties to fit inside single localadmin. That will produce what looks like locality dupes in search and for map display.

**Comment by stepps00** [comment]
> Are the localities often crossing the localadmin boundaries? Is this a multi-hierarchy issue?

There are only a handful of cases where a locality polygon overlaps these parent source geometries (counties).. and in each case, it looks like an issue with the locality polygon.. I will provide some examples/screenshots in a PR once I start updates in the repo.

**Comment by sindile** [comment]
I hope my comments below will make your life easy.

> **County (District Municipalities)**
> 
>     * No changes

**Here are the list of changes that are applicable, this is mainly due to name changes**

- Sarah Baartman (Cacadu - 1108730515)

- City of Johannesburg (Johannesburg - 1108730509) there is another Johannesburg - 102089581

- King Cetshwayo (Uthungulu - 1108730563)

- Harry Gwala (Sisonke - 1108730569

- Garden Route (Eden - 1108730621)


> **Localadmin (Local Municipalities)**
> 
>     * Replace existing WOF Point geometries with geometries from Local Municipalities file
> 
>     * In some cases, WOF records will be superseded (see examples in initial issue comment)

**New local municipalities were created by merging old ones**
- Alfred Duma <- Emnambithi/Ladysmith - 1125304399, Indaka - 1125271469

- Big Five Hlabisa <- The Big 5 False Bay - 1125271707, Hlabisa - 1125358625

- Collins Chabane <- Makhado - 1125271833, Thulamela - 1125271677

- Dawid Kruiper <- Mier - 1125421539 //Khara Hais - 1125271483

- Dr Beyers Naude <- Camdeboo - 1125271581 Ikwezi - 1125271811 Baviaans - 1125278525

- Dr Nkosazana Dlamini Zuma <- Ingwe - 1125285699 Kwa Sani - 1125271745

- Enoch Mgijima <- Tsolwana - 1125421547 Inkwanca - 1125271525 Lukanji - 1125374223

- Fetakgomo Tubatse <- Fetakgomo - 1125358593 Greater Tubatse - 1125271421

- Inkosi Langalibalele <- Imbabazane - 1125271409 Umtshezi - 1125356827

- JB Marks <- Tlokwe City Council - 1125421549 Ventersdorp 1125277589 / 1125382583

- Modimolle <- Mookgophong Modimolle - 1125358645 Mookgopong - 1125410011

- Rand West City <- Randfontein - 1125271723 Westonaria - 1125298727 / 1125374251

- Ray Nkonyeni <- Ezingoleni - 1125362675 Hibiscus Coast - 1125412501

- Walter Sisulu <- Maletswai - 1125271549 Gariep - 1125374195

**The name used differs from the official name - typo?**

- Chief Albert Luthuli <- Albert Luthuli - 1125285711

- City of Mbombela <- Mbombela - 1125320713

- eThekwini <- Ethekwini - 1125360055

- Sol Plaatje <- Sol Plaatjie - 1125271587

- uMuziwabantu <- UMuziwabantu - 1125271857

- uPhongolo <- UPhongolo - 1125421535


> @sindile - you mentioned the regions will need to be updated, but I'm actually seeing the Province data in these files matches the region data currently in WOF. Do you have examples where a WOF region needs an update?
There are no changes required here @stepps00 my mistake


> Also, I see unique codes for features in the Local Municipalities file, but I'm unable to find a Name attribute. Any idea where we'd find official names for these places?
 do you mean this data wards - https://dataportal-mdb-sa.opendata.arcgis.com/datasets/9092c804a70a49efa40932082eacdb55_0 @stepps00 
There are no names associated with them. These are called wards and are referred using the numbers under "WardNo" field.
As an example refer to the picture below: one would say I leave in Beacon Bay and the ward is Ward 28 in Buffalo City Municipality. They are used for electoral purposes - https://en.wikipedia.org/wiki/Ward_(South_Africa)
![image](https://user-images.githubusercontent.com/1027916/151367694-f6946889-f434-4a1f-98d4-c5d9e267eec1.png)

**Comment by stepps00** [comment]
Great, thanks for the additional info. I've updated my comment to call out some needed name changes at the county and localadmin placetypes.

> Do you mean this data wards? - https://dataportal-mdb-sa.opendata.arcgis.com/datasets/9092c804a70a49efa40932082eacdb55_0 @stepps00
There are no names associated with them. These are called wards and are referred using the numbers under "WardNo" field.
As an example refer to the picture below: one would say I leave in Beacon Bay and the ward is Ward 28 in Buffalo City Municipality. They are used for electoral purposes - https://en.wikipedia.org/wiki/Ward_(South_Africa)

Ah yes, this dataset. That makes sense that these are electoral wards.. we won't be using these to update WOF records, but they're a useful reference.

I think we've got all the needed data to proceed, so I'll flag you in a PR once one has been created.
